modules/openapi-generator/src/test/java/org/openapitools/codegen/html/StaticHtmlGeneratorTest.java

```diff
@@ -54,4 +56,18 @@ public void testSpecWithoutSchema() throws Exception {
         Assert.assertEquals(openAPI.getInfo().getTitle(), "ping test");
     }
 
+    @Test(description = "ensure that snake_case propery names wont be converted to snakeUnderscorecase")
+    public void testFromPropertyWithUnderscores() {
+        final Schema schema = new Schema()
+            .description("a sample model with property containing an underscore")
+            .addProperties("favorite_food", new StringSchema());
+        final OpenAPI openAPI = TestUtils.createOpenAPIWithOneSchema("UnderscoreTest", schema);
+        final DefaultCodegen codegen = new StaticHtmlGenerator();
+        codegen.setOpenAPI(openAPI);
+
+        CodegenProperty property = codegen.fromProperty("favorite_food", (Schema) openAPI.getComponents().getSchemas().get("UnderscoreTest").getProperties().get("favorite_food"));
+
+        Assert.assertEquals(property.baseName, "favorite_food");
+        Assert.assertEquals(property.name, "favorite_food");
+    }
```

Comment:
@farrez Thanks for adding the test case. Can the issue be also resolved by using `baseName` instead of `name` in the mustache templates as `baseName` should store the original property name defined in the spec.

Reply:
Yes that would also have worked.

Reply:
👌 I'll make the change with another PR.